arshal_default.go

```diff
@@ -1257,22 +1280,32 @@ func makeStructArshaler(t reflect.Type) *arshaler {
 				if len(f.index) > 1 {
 					v = v.fieldByIndex(f.index[1:], true)
 					if !v.IsValid() {
-						return newUnmarshalErrorBefore(dec, t, errNilField)
+						err := newUnmarshalErrorBefore(dec, t, errNilField)
+						if !uo.Flags.Get(jsonflags.ReportErrorsWithLegacySemantics) {
+							return err
+						}
+						errUnmarshal = cmp.Or(errUnmarshal, err)
+						unmarshal = func(dec *jsontext.Decoder, _ addressableValue, _ *jsonopts.Struct) error {
+							return dec.SkipValue()
+						}
```

Comment:
should we document these bits of legacy error logic? not the `cmp.Or`, but e.g. this SkipValue unmarshal, or the SkipUntil below. the logic isn't intricate, but looking at this, it's not obvious to me how or why the original error semantics were different.

Reply:
reading the rest of the PR, you do document this in the legacy error semantics option. Perhaps simply do a doc link to that.

Reply:
I'll expand documentation of how lazy errors work in a subsequent PR. I'm planning on adding a v2 option to make this accessible to v2-only users. When that happens, we'll use proper multi-errors and not this lame "first error wins" semantic that legacy supports.